test/Inputs/objc_direct.h

```diff
@@ -1,6 +1,7 @@
 #import <Foundation/Foundation.h>
 
 @interface Bar : NSObject
++ (instancetype)BarWithValue:(int)value __attribute__((objc_direct));
```

Comment:
Nit: Spelling it `barWithValue:` is a more realistic example of how this might appear in a header.

Also, just to check that your change will affect them too, you might want to include a test with an `- (instancetype)init...` initializer. Those may go down different paths in the importer, and we'd like to make sure they still hit this logic.